src/OpenLoco/src/Vehicles/Routing.cpp

```diff
@@ -683,19 +700,220 @@ namespace OpenLoco::Vehicles
 
         auto filterFunction = [&routingTransformData](const LocationOfInterest& interest) { return findOccupationByBlock(interest, routingTransformData); };
         auto transformFunction = [&routingTransformData](const LocationOfInterestHashMap& interestMap) { setSignalsOccupiedState(interestMap, routingTransformData); };
+        LocationOfInterestHashMap interestMap{ kSignalHashMapSize };
 
         findAllTracksFilterTransform(
-            loc, trackAndDirection, company, trackType, filterFunction, transformFunction);
+            interestMap,
+            TrackNetworkSearchFlags::unk0 | TrackNetworkSearchFlags::unk2,
+            loc,
+            trackAndDirection,
+            company,
+            trackType,
+            filterFunction,
+            transformFunction);
     }
 
     uint8_t sub_4A2A58(const World::Pos3& loc, const TrackAndDirection::_TrackAndDirection trackAndDirection, const CompanyId company, const uint8_t trackType)
     {
         // 0x001135F88
         uint16_t unk = 0;
-
+        LocationOfInterestHashMap interestMap{ kSignalHashMapSize };
         auto filterFunction = [&unk](const LocationOfInterest& interest) { return sub_4A2D4C(interest, unk); };
-        findAllTracksFilterTransform(loc, trackAndDirection, company, trackType, filterFunction, kNullTransformFunction);
+
+        findAllTracksFilterTransform(
+            interestMap,
+            TrackNetworkSearchFlags::unk0 | TrackNetworkSearchFlags::unk2,
+            loc,
+            trackAndDirection,
+            company,
+            trackType,
+            filterFunction,
+            kNullTransformFunction);
 
         return unk;
     }
+
+    // 0x004A5D94
+    static bool applyTrackModToTrack(const LocationOfInterest& interest, const uint8_t flags, LocationOfInterestHashMap* hashMap, uint8_t modSelection, uint8_t trackObjectId, uint8_t trackModObjectIds, currency32_t& totalCost, CompanyId companyId, bool& hasFailedAllPlacement)
+    {
+        // If called from routing add reverse direction of track
```

Comment:
why does routing need to calling this?

Reply:
I had originally planned for this function to be in a header and so if you are calling this function for single element mod placement you don't need to do the reverse track stuff. But as i haven't done that I should probably change this to `If not single element placement add reverse direction of track to hashMap so it isn't visited. Track mods are not direction specific.`